inc/deployfile.class.php

```diff
@@ -955,6 +964,30 @@ public function removeFileInRepo($sha512)
         return true;
     }
 
+    /**
+    * Return all subpart path
+    *
+    * @param string $sha512 sha512 of the file
+    * @return array
+    */
+    public function getFilePath($sha512)
+    {
+        if (!$this->checkPresenceFile($sha512)) {
+            return false;
+        }
+
+        $path = [];
+        $handle = fopen(PLUGIN_GLPI_INVENTORY_MANIFESTS_DIR . $sha512, "r");
+        if ($handle) {
+            while (($buffer = fgets($handle)) !== false) {
+                $path[] = PLUGIN_GLPI_INVENTORY_REPOSITORY_DIR . $this->getDirBySha512($buffer) . "/" . trim($buffer, "\n");
+            }
+            fclose($handle);
+        }
```

Comment:
Something should be added in logs when an error occurs.

```suggestion
        $handle = fopen(PLUGIN_GLPI_INVENTORY_MANIFESTS_DIR . $sha512, "r");
        $error = $handle === false;
        if ($handle) {
            while (($buffer = fgets($handle)) !== false) {
                $path[] = PLUGIN_GLPI_INVENTORY_REPOSITORY_DIR . $this->getDirBySha512($buffer) . "/" . trim($buffer, "\n");
            }
            if (!feof($fp)) {
                $error = true;
            }
            fclose($handle);
        }
        if ($error) {
            trigger_error(
                sprintf('Unable to read file %s manifest.', $sha512),
                E_USER_WARNING
            );
            return false;
        }
```